### ISSUE TITLE
Ingress gateway cert rotation does not respect updated mesh config settgings

### ISSUE THREAD

**Issue by allenlsy** [openservicemesh/osm#4398]
**Bug description**:

Ingress gateway cert is issued by osm-controller. When the cert validity duration is updated, osm-controller does not always respect to the updated value. It only issues the cert with the updated validity duration the first time afterwards.

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [X]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [X]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**:

osm-controller should always issue cert with updated validity duration afterwards.

**Steps to reproduce the bug (as precisely as possible)**:

2. Install Nginx ingress controller, following [this guide](https://kubernetes.github.io/ingress-nginx/) and [OSM ingress integration guide](https://release-v0-11.docs.openservicemesh.io/docs/demos/ingress_k8s_nginx/). Then ensure ingress controller namespace is added to OSM
3. Update control plane log level to `trace` by updating `observability.osmLogLevel` field in mesh config. Then start to tail the log of osm controller.
4. Update `certificate.ingressGateway.validityDuration` to `10m`. We should see that osm-controller issues  a new cert to ingress gateway with the validity duration of 10m.

![Screen Shot 2021-12-14 at 11 18 19 AM](https://user-images.githubusercontent.com/872876/146236928-a4b943f7-20ad-40c3-a4a1-d75ec08a66f3.png)

5. After the new cert expires, the next new ingress cert has validity duration of 24h, which is the original value.

![Screen Shot 2021-12-14 at 11 14 55 AM](https://user-images.githubusercontent.com/872876/146237182-74a0d4e6-a69c-4a42-a934-2cb1c3d14c48.png)

**How was OSM installed?**:

Create a cluster with OSM installed with default settings, e.g. follow the [installation guide](https://release-v0-11.docs.openservicemesh.io/docs/getting_started/install/)

**Anything else we need to know?**:

**Bug report archive**:

**Environment**:
- OSM version (use `osm version`): v0.11.0
- Kubernetes version (use `kubectl version`): v1.20.9
- Size of cluster (number of worker nodes in the cluster): 3
- Others:

**Comment by shashankram** [comment]
This was resolved on `main` (not backported to v0.11) by https://github.com/openservicemesh/osm/pull/4285.
![image](https://user-images.githubusercontent.com/21697719/146553739-d3cff6f6-c6b8-430a-8f51-6f6032cc748c.png)

**Comment by trstringer** [comment]
Ah that makes sense why I was unable to repro this behavior:

```
notBefore=Dec 16 22:56:30 2021 GMT
notAfter=Dec 16 23:01:30 2021 GMT
Thu 16 Dec 2021 05:57:02 PM EST - Setting ingress certificate from 5 minutes to 10 minutes
meshconfig.config.openservicemesh.io/osm-mesh-config patched
Thu 16 Dec 2021 05:57:03 PM EST - Sleeping for the certificate to expire and rotate...
Thu 16 Dec 2021 06:03:03 PM EST - Done sleeping
notBefore=Dec 16 22:57:03 2021 GMT
notAfter=Dec 16 23:07:03 2021 GMT
```
